### DISCUSSION TITLE
mandatory true/false

### DISCUSSION THREAD

**Discussion by joaociocca** [pls-rs/pls#83] [Help Q&A]
so I'm back at using this to see if I can fully replace my `ls` usage with it... but I feel the mandatory need to supply "true" to be  bit heavy. I mean, if I type `pls -g` shouldn't it be the same as `pls --grid true` or even just `pls --grid`?

I want to bring attention to the mandatory part. I think there may be times when setting to false may be useful, and so a `--grid false` would stand... but as a user, I don't understand the need for `true`.

**Comment by dhruvkb** [comment]
On this, I completely agree that providing `true` is excessively verbose. But the command line parsing library that `pls` uses ([`clap`](https://docs.rs/clap/)) either

1. requires the value to be mandatory
   
   This means that either `true` or `false` must be supplied after the flag name. E.g. `--grid true` or `--grid=false`.

2. [requires an `=` between the flag and value](https://docs.rs/clap/4.3.11/clap/struct.Arg.html#method.default_missing_value)

   This means a space is no longer a valid separator. E.g. `--grid` (equivalent to `--grid=true`) or `--grid=false`.

I chose 1 because that was easier to reason about. There might be workarounds for this so I'll definitely merge PRs fixing this or implement that myself if I find a solution.
